src/EnergyPlus/PurchasedAirManager.cc

```diff
@@ -179,6 +181,8 @@ namespace PurchasedAirManager {
     int NumPurchAir;
     int NumPlenumArrays;
     bool GetPurchAirInputFlag(true);
+    Array1D_bool EMSMdotSetupFlag;
+    Array1D_bool EMSTsupSetupFlag;
```

Comment:
In this branch you added two new variables to the purchased air data struct:
 - int EMSMassFlowRateOutVarNum;
 - int EMSSupplyTempOutVarNum;     

Why not just add two more:
- bool EMSMdotSetupFlag;
- bool EMSTsupSetupFlag;

That way you don't have to have these extra arrays hanging out and allocate them separately and what not.

Reply:
seems to me that the better fix is to not skip EMS overrides because of schedule or zero load.

Reply:
@Myoldmopar @EnergyArchmage If EMS is not skipped, the calling structure needs to be revised. Here is a section code in HVACManager:

        ManageEMS(emsCallFromAfterHVACManagers, anyEMSRan); // calling point
        ManageEMS(emsCallFromHVACIterationLoop, anyEMSRan); // calling point id

        // first explicitly call each system type with FirstHVACIteration,

        // Manages the various component simulations
        SimSelectedEquipment(state, SimAirLoopsFlag,
                             SimZoneEquipmentFlag,
                             SimNonZoneEquipmentFlag,
                             SimPlantLoopsFlag,
                             SimElecCircuitsFlag,
                             FirstHVACIteration,
                             SimWithPlantFlowUnlocked);

The change is to combine all bool arguments as one. If any one is true, all zone equipment, airloop and plant loop have to called to update node and EMS override values, regardless of convergence status in one of loop. Unfortunately, it will slow down. This isssu is not a bug and becomes a new feature.

Reply:
I agree, based on the discussions in https://github.com/NREL/EnergyPlus/issues/5788 if the user set EMS flow >0, it should happen.

Reply:
@lgu1234 I don't think you need to mess with those flags. Just changing the logic within the ideal loads function should be adequate. Other equipment with EMS overrides follow the same pattern, I think.

Reply:
@mjwitte If I don't skip EMS override, there are no changes in EMS outputs. Anyway, I an going to make changes in the ideal loads module only based on comments.

Reply:
No I'm confused. My expectation for this was to simply move the existing EMSoverride code to a place where it won't be skipped if there is no load. Maybe that isn't easy given how the function is structured, but it should be possible. Perhaps there needs to be a duplicate EMSOverride check is the system falls into the "off" block?

Reply:
@mjwitte Let me explain what I found during debug in HVACManager

The EMS is called first 
ManageEMS(emsCallFromHVACIterationLoop, anyEMSRan); // calling point id

Then SimSelectedEquipment is called with a set of bool arguments

    SimSelectedEquipment(state, SimAirLoopsFlag, SimZoneEquipmentFlag,.....)

If revision occurs in IdealLoads module only as a part of ZoneEquipment. When SimZoneEquipmentFlag (an argument of SimSelectedEquipment) is false, ZoneEquipment execution is bypassed. That means any EMS overrides in IdealLoads will be bypassed. The override values from EMSManager will remain as is. 

What I have done is that I introduce bypassfactor to skip EMSManager override and keep override values in IdealLoads as final values.

Reply:
We don't worry about that for any other components. If `SimZoneEquipmentFlag` is false, then nothing should be changing at that point in the zone equipment. The equipment should always get called at least once.